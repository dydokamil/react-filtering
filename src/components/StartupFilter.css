input {
  margin-bottom: 1rem;
  transition: all 1s;
  width: 20%;
  text-align: center;
}

input:focus {
  width: 100%;
}
