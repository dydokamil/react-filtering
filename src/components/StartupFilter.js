import React from 'react'
import { connect } from 'react-redux'
import { filterStartups } from '../actions/startups'
import './StartupFilter.css'

export class StartupFilter extends React.Component {
  constructor () {
    super()

    this.state = { filter: '' }
  }

  onFilterChange = event => {
    this.setState({ filter: event.target.value }, () => {
      this.props.filterStartups(this.state.filter)
    })
  }

  render () {
    return (
      <div>
        <input
          id="filter-input"
          placeholder="Filter..."
          onChange={this.onFilterChange}
        />
      </div>
    )
  }
}

export default connect(null, { filterStartups })(StartupFilter)
