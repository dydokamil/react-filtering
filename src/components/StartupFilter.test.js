import React from 'react'
import { StartupFilter } from './StartupFilter'
import { shallow } from 'enzyme'

describe('Startup Filter', () => {
  const mockFilterStartups = jest.fn()
  const props = { filterStartups: mockFilterStartups }
  const startupFilter = shallow(<StartupFilter {...props} />)
  const startupName = 'Some Company Name'

  it('renders the startup filter', () => {
    expect(startupFilter).toMatchSnapshot()
  })

  it('should render an input with an id of #filter-input', () => {
    expect(startupFilter.find('#filter-input').exists()).toBe(true)
  })

  it('should initialize `state`', () => {
    expect(startupFilter.state()).toEqual({ filter: '' })
  })

  it('should change state when user types into `#filter-input`', () => {
    const filterInput = startupFilter.find('#filter-input')
    filterInput.simulate('change', { target: { value: startupName } })

    expect(startupFilter.state().filter).toEqual(startupName)
  })

  it('dispatches an action when user types into the filter using `filterStartups()` from props', () => {
    startupFilter
      .find('#filter-input')
      .simulate('change', { target: { value: 'Some value' } })
    expect(mockFilterStartups).toHaveBeenCalledWith(startupName)
  })
})
