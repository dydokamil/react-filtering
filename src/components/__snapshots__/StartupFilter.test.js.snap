// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Startup Filter renders the startup filter 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartupFilter
    filterStartups={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        id="filter-input"
        onChange={[Function]}
        placeholder="Filter..."
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "id": "filter-input",
        "onChange": [Function],
        "placeholder": "Filter...",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          id="filter-input"
          onChange={[Function]}
          placeholder="Filter..."
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "filter-input",
          "onChange": [Function],
          "placeholder": "Filter...",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
